onnxruntime/core/common/cpuid_info.h

```diff
@@ -122,8 +122,16 @@ class CPUIDInfo {
 
   void X86Init();
 #elif defined(CPUIDINFO_ARCH_ARM)
-  // Now the following var is only used in ARM build, but later one we may expand the usage.
+// Now the following var is only used in ARM build, but later one we may expand the usage.
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
```

Comment:
if the code can be fixed so that we don't have an unused variable in certain compilations, that would be ideal.

otherwise, could you add a comment about why these suppressions are necessary? it may be more direct to just suppress the unused variable warning with these pragmas instead.